core/src/main/java/net/sourceforge/jnlp/util/WindowsDesktopEntry.java

```diff
@@ -145,51 +143,44 @@ public void createWindowsMenu() throws IOException {
         sl.saveTo(path + "/" + link);
         ul.saveTo(path + "/Uninstall " + link);
         // write shortcuts to list
-        manageShortcutList(ManageMode.A, path + "/" + link);
-        manageShortcutList(ManageMode.A, path + "/Uninstall " + link);
+        manageShortcutList(path + "/" + link);
+        manageShortcutList(path + "/Uninstall " + link);
     }
 
-    private void manageShortcutList(ManageMode mode, String path) throws IOException {
-        if (!getWindowsShortcutsFile().exists()) {
-            getWindowsShortcutsFile().createNewFile();
+    private void manageShortcutList(String path) throws IOException {
+        final File shortcutFile = getWindowsShortcutsFile();
+        if (!shortcutFile.exists() && !shortcutFile.createNewFile()) {
+            LOG.warn("could not create file for shortcut manager: {}", shortcutFile);
+            return;
         }
-        LOG.debug("Using WindowsShortCutManager {}", getWindowsShortcutsFile().toString());
-        if (ManageMode.A == mode) {
-            List<String> lines = null;
-            // if UTF-8 fails, try ISO-8859-1
-            try {
-                LOG.debug("Reading Shortcuts with UTF-8");
-                lines = Files.readAllLines(getWindowsShortcutsFile().toPath(), UTF_8);
-            } catch (MalformedInputException me) {
-                LOG.debug("Fallback to reading Shortcuts with default encoding {}", Charset.defaultCharset().name());
-                lines = Files.readAllLines(getWindowsShortcutsFile().toPath(), Charset.defaultCharset());
-            }
-            Iterator it = lines.iterator();
-            String sItem = "";
-            String sPath;
-            Boolean fAdd = true;
-            // check to see if line exists, if not add it
-            while (it.hasNext()) {
-                sItem = it.next().toString();
-                String[] sArray = sItem.split(",");
-                String application = sArray[0]; //??
-                sPath = sArray[1];
-                if (sPath.equalsIgnoreCase(path)) {
-                    // it exists don't add
-                    fAdd = false;
-                    break;
-                }
-            }
-            if (fAdd) {
-                LOG.debug("Default encoding is {}", Charset.defaultCharset().name());
-                LOG.debug("Adding Shortcut to list = {} with UTF-8 encoding", sItem);
-                String scInfo = file.getFileLocation().toString() + ",";
-                scInfo += path + "\r\n";
-                Files.write(getWindowsShortcutsFile().toPath(), scInfo.getBytes(UTF_8), StandardOpenOption.APPEND);
-            }
+        LOG.debug("Using WindowsShortCutManager {}", shortcutFile);
+        final List<String> lines = readAllLine(shortcutFile);
```

Comment:
wouldn't be "readAllLines" a better method name here?

Reply:
full ack